pkg/stanza/fileconsumer/internal/fingerprint/fingerprint.go

```diff
@@ -96,6 +96,10 @@ func (f *Fingerprint) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+func (f *Fingerprint) GetFingerprint() *Fingerprint {
+	return f
+}
+
```

Comment:
I've added this so the `Fingeprint` can implement the `Matchable` interface, so that we can have unified return type for https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ad46f94b11ea1592a33768eb332adc5b1d7cdc33/pkg/stanza/fileconsumer/internal/tracker/tracker.go#L194-L195

I think this is neater than returning `[]any` or creating a struct to capture the fingerprint/metadata.

Reply:
We don't need to return the fingerprints that didn't match. Just return `[]*reader.Metadata` and leave the not found items empty.